### PR TITLE
feat(core): Contentful entry tagging service

### PR DESCRIPTION
**Pull request by RunarVestmann**
# Contentful entry tagging service

## Why

* We want to manage user permissions in Contentful and the approach chosen was to utilize roles and tags.
* When a user creates an entry then this service will automatically apply the correct tag to that entry so the user permissions are correctly set (since the tag will be connected to the role that the user is linked to)

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: workspace.json [hunk 1]
```diff
@@ -9569,6 +9569,59 @@
...
+          "configurations": {
+            "production": {
+              "optimization": true,
+              "extractLicenses": true,
+              "inspect": false,
+              "fileReplacements": [
+                {
+                  "replace": "apps/services/contentful-entry-tagger/src/environments/environment.ts",
+                  "with": "apps/services/contentful-entry-tagger/src/environments/environment.prod.ts"
+                }
```
Comment (pshomov) [inline workspace.json:9597]: We stopped using this method of configuration.

**Review by pshomov — CHANGES_REQUESTED** [review]
Will add the docker image to our registry so it is ready to go.
Needs to have `infra` folder with a .ts file describing the deployment. For example, see other similar services. Also needs to be wired up in the [islandis chart/deployment](https://github.com/island-is/island.is/blob/main/infra/src/uber-charts/islandis.ts)

File: workspace.json [hunk 2]
```diff
@@ -9569,6 +9569,59 @@
...
+        },
+        "test": {
+          "builder": "@nrwl/jest:jest",
+          "outputs": ["coverage/apps/services/contentful-entry-tagger"],
+          "options": {
+            "jestConfig": "apps/services/contentful-entry-tagger/jest.config.js",
+            "passWithNoTests": true
+          }
+        }
+      }
```
[the remark this answers is not in the document]
Comment (pshomov) [reply workspace.json:9624]: This is so a docker image is created

File: workspace.json (hunk 1 above)
Comment (RunarVestmann) [reply workspace.json]: Should I remove the replace and with then? This got added automatically when I bootstrapped the service

Comment (pshomov) [reply workspace.json]: Just remove this to begin with. There is a library that is used to retrieve config if I recall correctly.

File: workspace.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (pshomov) [reply workspace.json]: Sorry, yes - that sounds good.

File: apps/services/contentful-entry-tagger/infra/contentful-entry-tagger-service.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { service, ServiceBuilder } from '../../../../infra/src/dsl/dsl'
+
+export const serviceSetup = (): ServiceBuilder<'contentful-entry-tagger-service'> =>
+  service('contentful-entry-tagger-service')
+    .image('contentful-entry-tagger')
```
Comment (pshomov) [inline apps/services/contentful-entry-tagger/infra/contentful-entry-tagger-service.ts:5]: Please use the name as defined in `workspace.json`

File: workspace.json (hunk 1 above)
Comment (RunarVestmann) [reply workspace.json]: Done :)

File: infra/src/uber-charts/islandis.ts [hunk 4]
```diff
@@ -168,4 +171,5 @@ export const FeatureDeploymentServices = []
 export const ExcludedFeatureDeploymentServices = [
   userNotificationService,
   userNotificationWorkerService,
+  contentfulEntryTagger,
```
Comment (pshomov) [inline infra/src/uber-charts/islandis.ts:174]: Just FYI - this means the service cannot be feature-deployed.

File: infra/src/uber-charts/islandis.ts [hunk 5]
```diff
@@ -108,6 +110,7 @@ export const Services: EnvironmentServices = {
     appSystemApiWorker,
     userNotificationService,
     userNotificationWorkerService,
+    contentfulEntryTagger,
```
Comment (pshomov) [inline infra/src/uber-charts/islandis.ts:113]: You should probably start by having the service added only to the dev stack, while it is being developed

File: apps/services/contentful-entry-tagger/infra/contentful-entry-tagger-service.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    .resources({
+      requests: {
+        cpu: '100m',
+        memory: '256Mi',
+      },
+      limits: {
+        cpu: '400m',
+        memory: '1024Mi',
+      },
+    })
```
Comment (pshomov) [inline apps/services/contentful-entry-tagger/infra/contentful-entry-tagger-service.ts:21]: No change needed - You can skip this part if the service has no particular needs.

File: apps/services/contentful-entry-tagger/infra/contentful-entry-tagger-service.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { service, ServiceBuilder } from '../../../../infra/src/dsl/dsl'
+
+export const serviceSetup = (): ServiceBuilder<'contentful-entry-tagger-service'> =>
+  service('contentful-entry-tagger-service')
```
Comment (pshomov) [inline apps/services/contentful-entry-tagger/infra/contentful-entry-tagger-service.ts:4]: This name has to match the one in the [helm repo](https://github.com/island-is/helm/pull/645).

**Review by pshomov — CHANGES_REQUESTED** [review]

File: apps/services/contentful-entry-tagger/infra/contentful-entry-tagger-service.ts (hunk 7 above)
Comment (RunarVestmann) [reply apps/services/contentful-entry-tagger/infra/contentful-entry-tagger-service.ts]: I changed the name in the helm repo to be contentful-entry-tagger-service

File: infra/src/uber-charts/islandis.ts (hunk 5 above)
Comment (RunarVestmann) [reply infra/src/uber-charts/islandis.ts]: Ok, yep I agree

File: infra/src/uber-charts/islandis.ts (hunk 4 above)
Comment (RunarVestmann) [reply infra/src/uber-charts/islandis.ts:174]: That's fine, I don't see a reason for it to be feature deployed

File: apps/services/contentful-entry-tagger/infra/contentful-entry-tagger-service.ts (hunk 3 above)
Comment (RunarVestmann) [reply apps/services/contentful-entry-tagger/infra/contentful-entry-tagger-service.ts]: I think I've got matching names now

**Review by Snojo — APPROVED** [review]
LGTM on the DevOps Side

**Comment by RunarVestmann** [comment]
As of right now the ingress is open so it can be called from anywhere, I could look into request verification to make sure the request is coming from Contentful.

**Pull request merged**